src/Spectre.Data/Datasets/IDataset.cs

```diff
@@ -0,0 +1,146 @@
+﻿/*
+ * IDataset.cs
+ * Contains interface class for basic functionalities of dataset 
+ * representing measurements from single sample point
+ * 
+   Copyright 2017 Dariusz Kuchta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System.Collections.Generic;
+using Spectre.Data.Structures;
+
+namespace Spectre.Data.Datasets
+{
+    public interface IDataset
+    {
+        //TODO: Somehow divide the available functionalities to the
+        //TODO: external world (DataPoint accessible from higher
+        //TODO: levels of abstraction, raw data for algorithms).
+        //TODO: ...
+        //TODO: Derive a sub-interface?
```

Comment:
Similarly as above. Issues => Trello. Nevertheless, good point (on interface). Not necessarily sub-interface, but idea is worth trying.